### PR TITLE
Implemented blogposting for users

### LINKED ISSUE #13: Create Blogpost Form
The users currently can post comments, but not blogposts. Create a form for users to post their own blogposts.

### PR DESCRIPTION
**Pull request by stevenpi**
As requested in the Issue, users can now post their own blogpost

Resolves #13

### CODE REVIEW

File: Blog/views.py [hunk 1]
```diff
@@ -50,6 +51,28 @@ def add_comment(request, pk):
...
+def add_post_form(request):
+    do_add_url = reverse('post-add-do')
+    context = {'do_add_url': do_add_url}
+    return render(request, 'Blog/post_add.html', context)
+
+
+@login_required
+def add_post_do_add(request):
+    if request.method == "POST":
+        content = request.POST.get("content", "")
```
Comment (mbaechtold) [inline Blog/views.py]: You are not validating the user input at all. This opens your website to some nasty attacks (XSS, etc).

You could use a class based `CreateView` for this kind of task in combination with a `ModelForm`.

**Comment by stevenpi** [comment]
Improved the code as requested, by using a class-based form

File: Blog/views.py [hunk 2]
```diff
@@ -65,6 +55,27 @@ def post_detail_view(request, pk):
...
+    paginate_by = 10
+
+
+class UserListView(generic.ListView):
+    model = User
+    paginate_by = 10
+    template_name = 'Blog/user_list.html'
+
+
+class PostCreate(generic.CreateView, LoginRequiredMixin):
```
[the remark this answers is not in the document]
Comment (mbaechtold) [reply Blog/views.py]: Before we talk about the MRO, please read https://django-advanced-training.readthedocs.io/en/latest/features/class-based-views/

[the remark this answers is not in the document]
Comment (mbaechtold) [reply Blog/views.py]: And https://docs.djangoproject.com/en/2.0/topics/class-based-views/intro/ is also a good read to better know the execution flow in class based views.

**Comment by stevenpi** [comment]
Resolved all the flaws mentioned in the review.
I was able to fix the login issue, however I think I still need some explanation about `MRO`

**Pull request merged**